File: game/scripts/vscripts/modifiers/generic/modifier_visible.lua
modifier_visible = class({})

function modifier_visible:IsHidden()        return false end
function modifier_visible:IsDebuff()        return	false	end
function modifier_visible:IsPurgable() 		return	false	end
function modifier_visible:IsPermanent()   return	false	end
function modifier_visible:RemoveOnDeath() 	return	false	end

File: src/vscripts/modifiers/modifier_visible.ts
import { CustomModifier } from "../abilities/framework/custom_modifier";
import { registerModifier } from "../lib/dota_ts_adapter";

@registerModifier("modifier_visible")
export class ModifierVisible extends CustomModifier<undefined> {
    IsHidden() {
        return true;
    }

    IsDebuff() {
        return false;
    }

    IsPurgable() {
        return false;
    }

    IsPermanent() {
        return false;
    }

    RemoveOnDeath() {
        return false;
    }
}
